jupyterhub/files/hub/jupyterhub_config.py

```diff
@@ -163,8 +163,17 @@ def camelCaseify(s):
 
     c.KubeSpawner.image = image
 
-if get_config('singleuser.imagePullSecret.enabled'):
-    c.KubeSpawner.image_pull_secrets = 'singleuser-image-credentials'
+# Combine global.imagePullSecret.create (single), global.imagePullSecrets
+# (list), and singleuser.image.pullSecrets (list).
+image_pull_secrets = []
+if get_config('global.imagePullSecret.create') or get_config('global.imagePullSecret.enabled'):
+    image_pull_secrets.append('image-registry-credentials')
+if get_config('global.imagePullSecrets'):
+    image_pull_secrets.extend(get_config('global.imagePullSecrets'))
+if get_config('singleuser.image.pullSecrets'):
+    image_pull_secrets.extend(get_config('singleuser.image.pullSecrets'))
+if image_pull_secrets:
+    c.KubeSpawner.image_pull_secrets = image_pull_secrets
```

Comment:
This is the logic that configures KubeSpawner in the same way that other pods are configured through a Helm chart helper function defined in `_helpers.tpl`, but KubeSpawner must be configured here with different syntax.